### ISSUE TITLE
dates test segfaults on arm

### ISSUE THREAD

**Issue by ViralBShah** [JuliaLang/julia#12355]
The dates test segfaults on arm. Using llvm 3.6.1

````
(precise)viral@localhost:~$ ./julia/julia
               _
   _       _ _(_)_     |  A fresh approach to technical computing
  (_)     | (_) (_)    |  Documentation: http://docs.julialang.org
   _ _   _| |_  __ _   |  Type "help()" for help.
  | | | | | | |/ _` |  |
  | | |_| | | | (_| |  |  Version 0.4.0-dev+6253 (2015-07-26 15:40 UTC)
 _/ |\__'_|_|_|\__'_|  |  Commit a97b24b* (1 day old master)
|__/                   |  arm-linux-gnueabihf

julia> 
````

**Comment by Keno** [comment]
Does it segfault if you just run `adjusters.jl` and `io.jl` manually on the command line. If so, I may have a lead.

**Comment by ViralBShah** [comment]
Running manually on the command line, `io` does segfault, but `adjusters` does not.

````
(gdb) set args runtests.jl dates/io
(gdb) r
Starting program: /home/viral/julia/usr/bin/julia-debug runtests.jl dates/io
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib/arm-linux-gnueabihf/libthread_db.so.1".
     * dates/io             
Program received signal SIGSEGV, Segmentation fault.
__GI_____strtod_l_internal (nptr=0x3 <error: Cannot access memory at address 0x3>, endptr=0xbeffdc14, 
    group=<optimized out>, loc=0xb602243c <_nl_C_locobj>) at strtod_l.c:608
608     strtod_l.c: No such file or directory.
(gdb) bt
#0  __GI_____strtod_l_internal (nptr=0x3 <error: Cannot access memory at address 0x3>, endptr=0xbeffdc14, 
    group=<optimized out>, loc=0xb602243c <_nl_C_locobj>) at strtod_l.c:608
#1  0xb61c66fa in strtod_c (nptr=0x3 <error: Cannot access memory at address 0x3>, endptr=0xbeffdc14) at strtod.c:33
#2  0xb60fb9b4 in jl_try_substrtod (str=0x0, offset=3, len=3054483733) at builtins.c:809
#3  0x930b67cc in julia_parse_20892 () at parse.jl:163
#4  0x930b65f4 in julia_slotparse_20891 () at dates/io.jl:101
#5  0x930b61d8 in julia_getslot_20890 (cursor=<optimized out>) at dates/io.jl:108
#6  0x930b6398 in jlcall_getslot_20890 ()
#7  0xb60f8fa4 in jl_apply (f=0x9603b1f0, args=0xbeffde50, nargs=4) at julia.h:1263
#8  0xb60fc104 in jl_trampoline (F=0x9603b1f0, args=0xbeffde50, nargs=4) at builtins.c:979
#9  0xb60efe38 in jl_apply (f=0x9603b1f0, args=0xbeffde50, nargs=4) at julia.h:1263
#10 0xb60f3d8c in jl_apply_generic (F=0x93fee3b0, args=0xbeffde50, nargs=4) at gf.c:1675
#11 0x931bd3e8 in julia_parse_20729 (x=<optimized out>, df=<optimized out>) at dates/io.jl:122
#12 0x931bd0d8 in julia_call_20728 ()
#13 0xb60efe38 in jl_apply (f=0x946a82f0, args=0xbeffdf70, nargs=3) at julia.h:1263
#14 0xb60f3ce6 in jl_apply_generic (F=0x94e4f930, args=0xbeffdf70, nargs=3) at gf.c:1651
#15 0x930ba0b8 in julia_anonymous_20888 () at test.jl:90
#16 0xb60f8fa4 in jl_apply (f=0x96039810, args=0x0, nargs=0) at julia.h:1263
#17 0xb60fc104 in jl_trampoline (F=0x96039810, args=0x0, nargs=0) at builtins.c:979
#18 0x931ff0f4 in julia_do_test_20690 (body=<optimized out>, qex=<optimized out>) at test.jl:50
#19 0xb60efe38 in jl_apply (f=0x95da5570, args=0xbeffe2b4, nargs=2) at julia.h:1263
#20 0xb60f3ce6 in jl_apply_generic (F=0x94a1aa20, args=0xbeffe2b4, nargs=2) at gf.c:1651
#21 0xb617f0c0 in jl_apply (f=0x94a1aa20, args=0xbeffe2b4, nargs=2) at julia.h:1263
#22 0xb617f486 in do_call (f=0x94a1aa20, args=0x95aa39b4, nargs=2, eval0=0x0, locals=0x0, nl=0, ngensym=0)
    at interpreter.c:65
#23 0xb617fef6 in eval (e=0x96039720, locals=0x0, nl=0, ngensym=0) at interpreter.c:212
#24 0xb617f1ec in jl_interpret_toplevel_expr (e=0x96039720) at interpreter.c:27
#25 0xb6197596 in jl_toplevel_eval_flex (e=0x960394d0, fast=1) at toplevel.c:524
#26 0xb6197898 in jl_parse_eval_all (fname=0x95d32790 "/home/viral/julia/test/dates/io.jl", len=34) at toplevel.c:574
#27 0xb6197a82 in jl_load (fname=0x95d32790 "/home/viral/julia/test/dates/io.jl", len=34) at toplevel.c:614
#28 0xb6197af4 in jl_load_ (str=0x95d72dd0) at toplevel.c:622
#29 0xb4997aa0 in julia_include_6828 (fname=<optimized out>) at boot.jl:254
#30 0xb60efe38 in jl_apply (f=0x93eb0850, args=0xbeffeb8c, nargs=1) at julia.h:1263
#31 0xb60f3ce6 in jl_apply_generic (F=0x93eb0780, args=0xbeffeb8c, nargs=1) at gf.c:1651
#32 0xb4ac27d8 in julia_include_from_node1_18574 (path=<optimized out>) at loading.jl:197
#33 0xb60efe38 in jl_apply (f=0x94b94720, args=0xbeffec54, nargs=1) at julia.h:1263
---Type <return> to continue, or q <return> to quit---
#34 0xb60f3ce6 in jl_apply_generic (F=0x94b946a0, args=0xbeffec54, nargs=1) at gf.c:1651
#35 0x9320f1dc in julia_runtests_20679 () at /home/viral/julia/test/testdefs.jl:196
#36 0x9320f3e8 in jlcall_runtests_20679 ()
#37 0xb60f8fa4 in jl_apply (f=0x95b8e7b0, args=0xbeffed80, nargs=1) at julia.h:1263
#38 0xb60fc104 in jl_trampoline (F=0x95b8e7b0, args=0xbeffed80, nargs=1) at builtins.c:979
#39 0xb60efe38 in jl_apply (f=0x95b8e7b0, args=0xbeffed80, nargs=1) at julia.h:1263
#40 0xb60f3d8c in jl_apply_generic (F=0x95b30320, args=0xbeffed80, nargs=1) at gf.c:1675
#41 0xb60f8fa4 in jl_apply (f=0x95b30320, args=0xbeffed80, nargs=1) at julia.h:1263
#42 0xb60fa79a in jl_f_apply (F=0x0, args=0xbeffee14, nargs=2) at builtins.c:485
#43 0x93215098 in julia_anonymous_20676 () at multi.jl:658
#44 0xb60f8fa4 in jl_apply (f=0x95b8e720, args=0x0, nargs=0) at julia.h:1263
#45 0xb60fc104 in jl_trampoline (F=0x95b8e720, args=0x0, nargs=0) at builtins.c:979
#46 0x9324f208 in julia_run_work_thunk_20640 (thunk=<optimized out>) at multi.jl:619
#47 0x932190a0 in julia_remotecall_fetch_20674 (w=<optimized out>, f=<optimized out>) at multi.jl:692
#48 0xb60f8fa4 in jl_apply (f=0x95b8d050, args=0xbefff180, nargs=3) at julia.h:1263
#49 0xb60fc104 in jl_trampoline (F=0x95b8d050, args=0xbefff180, nargs=3) at builtins.c:979
#50 0xb60efe38 in jl_apply (f=0x95b8d050, args=0xbefff180, nargs=3) at julia.h:1263
#51 0xb60f3d8c in jl_apply_generic (F=0x94a1a140, args=0xbefff180, nargs=3) at gf.c:1675
#52 0xb60f8fa4 in jl_apply (f=0x94a1a140, args=0xbefff180, nargs=3) at julia.h:1263
#53 0xb60fa79a in jl_f_apply (F=0x0, args=0xbefff220, nargs=3) at builtins.c:485
#54 0x9321b0d4 in julia_remotecall_fetch_20673 (id=0, f=<optimized out>) at multi.jl:707
#55 0xb60f8fa4 in jl_apply (f=0x95b50770, args=0xbefff340, nargs=3) at julia.h:1263
#56 0xb60fc104 in jl_trampoline (F=0x95b50770, args=0xbefff340, nargs=3) at builtins.c:979
#57 0xb60efe38 in jl_apply (f=0x95b50770, args=0xbefff340, nargs=3) at julia.h:1263
#58 0xb60f3d8c in jl_apply_generic (F=0x94a1a140, args=0xbefff340, nargs=3) at gf.c:1675
#59 0xb60f8fa4 in jl_apply (f=0x94a1a140, args=0xbefff340, nargs=3) at julia.h:1263
#60 0xb60fa79a in jl_f_apply (F=0x0, args=0xbefff5d4, nargs=3) at builtins.c:485
#61 0x93234298 in julia_anonymous_20656 () at task.jl:1395
#62 0xb60f8fa4 in jl_apply (f=0x95806b50, args=0x0, nargs=0) at julia.h:1263
#63 0xb60fc104 in jl_trampoline (F=0x95806b50, args=0x0, nargs=0) at builtins.c:979
#64 0xb6189044 in jl_apply (f=0x95806b50, args=0x0, nargs=0) at julia.h:1263
#65 0xb6189564 in start_task () at task.c:232
#66 0xb61895aa in set_base_ctx (__stk=0xbefff6db "") at task.c:241
#67 0xb6189606 in julia_init (rel=JL_IMAGE_JULIA_HOME) at task.c:256
#68 0x0000a8ce in main (argc=2, argv=0xbefff868) at repl.c:583
````

**Comment by ViralBShah** [comment]
Seems like the same issue as #12347

**Comment by yuyichao** [comment]
Should be fixed by #14194 now there could be other issues that make the strings test fail but that would be a different issue.
